### ISSUE TITLE
Query metadata in EXIF

### ISSUE THREAD

**Issue by narensankar0529** [RhetTbull/osxphotos#323]
Is there anyway to do a query on the database to locate photos with specific EXIF metadata tags?

For example EXIF:Software? To find photos that have been edited by a specific application

**Comment by RhetTbull** [comment]
No.  You can query using the `{exiftool:}` template but only for writing the resulting value out to sidecar or other EXIF data.  This would be a useful feature though and not difficult to implement.

**Comment by narensankar0529** [comment]
Yeah - This would be a cool feature. I am starting to find that some third party apps that can be use to edit Photos are writing bad exif data. When I started to export via osxphotos, exiftool starts to complain a lot. So it would be great to find all fo such pictures and hopefully then fix up the exif data. I couldn't find a way to search within Photos itself since it is very limited in what can be used as a search field.

**Comment by RhetTbull** [comment]
Maybe I answered too quickly.  I meant "not with osxphotos CLI".  Do you know any python? If so, this would be a short script using the osxphotos scripting interface.  Something like this (not tested!):
```python
import osxphotos

photosdb = osxphotos.PhotosDB()
for photo in photosdb.photos():
    exif = photo.exiftool
    if exif:
        # exiftool returns None if photo not present on disk
        exifdict = exif.asdict()
        if "EXIF:Software" in exifdict:
            # exiftool.asdict() uses group names for all tags
            print(f"{photo.original_filename}, {exifdict['EXIF:Software']} {photo.path}")
```

**Comment by RhetTbull** [comment]
If you did this in a script, you could also use my other python library [PhotoScript](https://github.com/RhetTbull/PhotoScript) to add all these images to a new Album in Photos for easy processing.  Both photoscript and osxphotos id photos using a universally unique identifier (UUID) (PhotoInfo.uuid in osxphotos) for easy interoperability.

**Comment by RhetTbull** [comment]
Add `--exif` to query options.  This would work for both `query` and `export`.  Format would be something like:

`osxphotos query --exif EXIF:Subject "Adobe Photoshop"`

To be useful, should return partial matches as the actual string for Photoshop looks something like this: `Adobe Photoshop CC 2014 (Windows)`

These queries would be slow as exiftool would need to be used to read exif data from each file.  Photos stores a subset of the EXIF data in `ZEXTENDEDATTRIBUTES` but it's only the camera information.

**Comment by RhetTbull** [comment]
Here's a more complete example that creates new albums for each different value of EXIF:Software and adds the photos to the albums.

```python
import osxphotos
import photoscript
from osxphotos.exiftool import ExifTool

ALBUM_TEMPLATE = "Software: {}{}"


def main():
    photosdb = osxphotos.PhotosDB()
    for photo in photosdb.photos():
        if photo.hasadjustments and photo.path_edited:
            # get the edited version
            exif_edited = ExifTool(photo.path_edited)
        else:
            exif_edited = None

        if exif_edited:
            process_photo(photo, exif_edited, edited=True)

        exif_original = photo.exiftool
        if exif_original:
            process_photo(photo, exif_original, edited=False)


def process_photo(photo, exif, edited):
    # PhotoInfo.exiftool returns None if photo not present on disk
    exifdict = exif.asdict()
    if "EXIF:Software" in exifdict:
        # exiftool.asdict() uses group names for all tags
        photoslib = photoscript.PhotosLibrary()
        album_type = "(edited) " if edited else "(original) "
        album_name = ALBUM_TEMPLATE.format(album_type, exifdict["EXIF:Software"])
        if album_name not in photoslib.album_names():
            album = photoslib.create_album(album_name)
        else:
            album = photoslib.album(album_name)

        print(
            f"Adding {photo.original_filename} {exifdict['EXIF:Software']} edited:{'yes' if edited else 'no'} to album {album_name}"
        )
        album.add([photoscript.Photo(photo.uuid)])


if __name__ == "__main__":
    main()
```

**Comment by RhetTbull** [comment]
Fixed Issue #324 which will allow me to implement query based on exiftool.

**Comment by RhetTbull** [comment]
Implemented in current version on Github, will add to next release
